src/Function.h

```diff
@@ -137,6 +137,11 @@ class Function {
     /** Get the name of the function. */
     EXPORT const std::string &name() const;
 
+    /** If this is a wrapper of another func, created by a chain of in
+     * or clone_in calls, returns the name of the original
+     * Func. Otherwise returns the name. */
+    EXPORT const std::string &parent_name() const;
```

Comment:
So here's a question I don't know the answer to: Do we want to just track the name, or do we want to return the Func?